url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/entities/appliances/ConsumptionModel.java

```diff
@@ -0,0 +1,358 @@
+	 * @return the i-th pattern's duration
+	 */
+	public int getPatternDuration(int i) { return patternDuration[i]; }
+
+	/**
+	 * Gets the i-th pattern.
+	 *
+	 * @param i the index i
+	 * @return the i-th pattern
+	 */
+	public ArrayList<Tripplet> getPattern(int i) { return patterns[i]; }
+
+	/**
+	 * Gets the array storing the consumption patterns.
+	 *
+	 * @return the array storing the consumption patterns
+	 */
+	public ArrayList<Tripplet>[] getPatterns() { return patterns; }
+
+	/**
+	 * Check if consumption is static.
+	 *
+	 * @return true, if consumption is static.
+	 */
+	public boolean checkStatic()
+	{
+		boolean result = true;
+
+		Double[] values = getValues();
+
+		// System.out.println("Appliance: " + name + " Model: "
+		// + activeConsumptionModelString);
+		//
+		// System.out.println("Appliance: " + name + " Values: "
+		// + Arrays.toString(values));
+
+		for (int i = 0; i < values.length - 1; i++) {
+			// System.out.println("Previous: " + values[i].doubleValue() + " Next: "
+			// + values[i + 1].doubleValue());
+			if (values[i].doubleValue() != values[i + 1].doubleValue()) {
+				// System.out.println("IN");
+				result = false;
+				break;
+			}
+		}
+
+		return result;
+	}
+
+	/**
+	 * Gets the consumption.
+	 *
+	 * @return the consumption
+	 */
+	public Double[] getConsumption()
+	{
+
+		ArrayList<Double> temp = new ArrayList<Double>();
+		int times = getOuterN();
+		if (times == 0)
+			times = 2;
+		// Number of repeats
+		for (int i = 0; i < times; i++) {
+			// System.out.println("Time: " + i);
+			// Number of patterns in each repeat
+			for (int j = 0; j < getPatternN(); j++) {
+				// System.out.println("Pattern: " + j);
+				int internalTimes = getN(j);
+				if (internalTimes == 0)
+					internalTimes = 2;
+				// System.out.println("Internal Times: " + k);
+				for (int k = 0; k < internalTimes; k++) {
+					ArrayList<Tripplet> tripplets = getPattern(j);
+					for (int l = 0; l < tripplets.size(); l++) {
+						// System.out.println("TripletPower: " + l);
+						for (int m = 0; m < tripplets.get(l).d; m++) {
+							temp.add(tripplets.get(l).v);
+						}
+					}
+				}
+			}
+		}
+		Double[] result = new Double[temp.size()];
+		temp.toArray(result);
+		return result;
+
+	}
+
+	/**
+	 * Gets the consumption model as a BasicDBObject.
+	 *
+	 * @return the consumption model as a BasicDBObject
+	 */
+	public BasicDBObject toDBObject() {
+		BasicDBObject obj = new BasicDBObject();
+		obj.put("name", name);
+		obj.put("description", description);
+		obj.put("app_id", parentId);
+		obj.put("model", model);
+		return obj;
+	}
+
+
+	private void init (DBObject modelObj, String type) throws Exception {
+
+		try {
+
+			try {
+				outerN = ((Integer)modelObj.get("n")).intValue();
+			} catch(ClassCastException e ) {
+				try {
+					outerN = ((Long)modelObj.get("n")).intValue();
+				} catch(ClassCastException e1 ) {
+					outerN = ((Double)modelObj.get("n")).intValue();
+				}
+			} catch(NullPointerException npe) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 263, 280, 292, 306.